examples/platforms/posix/ble_nimble.c

```diff
@@ -0,0 +1,1111 @@
+    {
+        advp.conn_mode = BLE_GAP_CONN_MODE_NON;
+    }
+    if (aType | OT_BLE_ADV_MODE_SCANNABLE)
+    {
+        advp.disc_mode = BLE_GAP_DISC_MODE_GEN;
+    }
+    else
+    {
+        advp.disc_mode = BLE_GAP_DISC_MODE_NON;
+    }
+    rc = ble_gap_adv_start(DEFAULT_ADDR_TYPE, NULL, BLE_HS_FOREVER, &advp, gap_event_cb, aInstance);
+
+    return mapNimbleToOtError(rc);
+}
+
+otError otPlatBleGapAdvStop(otInstance *aInstance)
+{
+    int rc = ble_gap_adv_stop();
+
+    (void)aInstance;
+
+    return mapNimbleToOtError(rc);
+}
+
+otError otPlatBleGapScanStart(otInstance *aInstance, uint16_t aInterval, uint16_t aWindow)
+{
+    (void)aInstance;
+
+    struct ble_gap_disc_params discParams;
+    discParams.itvl              = aInterval;
+    discParams.window            = aWindow;
+    discParams.passive           = 1;
+    discParams.limited           = 0;
+    discParams.filter_policy     = 0;
+    discParams.filter_duplicates = 0;
+
+    int rc = ble_gap_disc(BLE_ADDR_PUBLIC, BLE_HS_FOREVER, &discParams, gap_event_cb, aInstance);
+
+    return mapNimbleToOtError(rc);
+}
+
+otError otPlatBleGapScanStop(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    int rc = ble_gap_disc_cancel();
+
+    return mapNimbleToOtError(rc);
+}
+
+otError otPlatBleGapConnect(otInstance *         aInstance,
+                            otPlatBleDeviceAddr *aAddress,
+                            uint16_t             aScanInterval,
+                            uint16_t             aScanWindow)
+{
+    (void)aInstance;
+
+    int                        rc;
+    ble_addr_t                 peerAddr;
+    struct ble_gap_conn_params connParams;
+    connParams.scan_itvl           = aScanInterval;
+    connParams.scan_window         = aScanWindow;
+    connParams.itvl_min            = 40;
+    connParams.itvl_max            = 56;
+    connParams.latency             = 0;
+    connParams.supervision_timeout = OT_BLE_CONN_SUPERVISOR_TIMEOUT_MAX;
+    connParams.min_ce_len          = 0;
+    connParams.max_ce_len          = 0;
+
+    mapOtToNimbleAddress(aAddress, &peerAddr);
+
+    rc = ble_gap_connect(aAddress->mAddrType, &peerAddr, DEFAULT_CONN_DISC_INTERVAL,
+                         &connParams, gap_event_cb, aInstance);
+
+    return mapNimbleToOtError(rc);
+}
+
+otError otPlatBleGapDisconnect(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    int rc;
+    rc = ble_gap_terminate(sNimbleConnHandle, BLE_ERR_REM_USER_CONN_TERM);
+    return mapNimbleToOtError(rc);
+}
+
+//=============================================================================
+//                        GATT COMMON
+//=============================================================================
+
+/**
+ * Registers vendor specific UUID Base.
+ *
+ * @param[in]  aInstance  The OpenThread instance structure.
+ * @param[in]  aUUID      A pointer to vendor specific 128-bit UUID Base.
+ *
+ */
+// otError otPlatBleGattVendorUuidRegister(otInstance *aInstance, const otPlatBleUuid *aUuid);
```

Comment:
Wouldn't be better to return OT_ERROR_NOT_IMPLEMENTED or even OT_ERROR_NONE. This functionality is not needed for this stack.